src/components/calculator.js

```diff
@@ -13,38 +20,119 @@ function Button(props) {
 Button.propTypes = {
   digit: PropTypes.string.isRequired,
   className: PropTypes.string,
+  handleClick: PropTypes.func.isRequired,
 };
 
 Button.defaultProps = {
   className: '',
 };
 
 function Calculator() {
+  const [previousOperand, setPreviousOperand] = useState('');
+  const [currentOperand, setCurrentOperand] = useState('0');
+  const [operation, setOperation] = useState('');
+
+  const handleDigitClick = (digit) => {
+    if (digit === '.' && currentOperand.includes('.')) return;
+    if (currentOperand === '0' && digit !== '.') {
+      setCurrentOperand(digit);
+    } else {
+      setCurrentOperand(currentOperand + digit);
+    }
+  };
+
+  const handleOperationClick = (operation) => {
+    if (previousOperand === '') {
+      setPreviousOperand(currentOperand);
+      setCurrentOperand('0');
+      setOperation(operation);
+    } else {
+      const result = operate(previousOperand, currentOperand, operation);
+      setPreviousOperand(result);
+      setCurrentOperand('0');
+      setOperation(operation);
+    }
+  };
+
+  const handleEqualsClick = () => {
+    const result = operate(previousOperand, currentOperand, operation);
+    setPreviousOperand('');
+    setCurrentOperand(result);
+    setOperation('');
+  };
+
+  const handleAllClearClick = () => {
+    setPreviousOperand('');
+    setCurrentOperand('0');
+    setOperation('');
+  };
+
+  const handleNegateClick = () => {
+    setCurrentOperand(Big(currentOperand).times(-1).toString());
+  };
+
+  const handlePercentageClick = () => {
+    setCurrentOperand(Big(currentOperand).div(100).toString());
+  };
```

Comment:
- [ ] Kindly fix the operation of `%`. In this project's scope, it means the modules sign not the percent sign.
### Hint
There is no need for a particular case to handle this operation as it's already handled in the provided files `calculate.js`, and `operate.js`.